test/framework/cluster_proxy.go

```diff
@@ -176,6 +181,18 @@ func (p *clusterProxy) GetScheme() *runtime.Scheme {
 	return p.scheme
 }
 
+func (p *clusterProxy) GetCache(namespace string, labelSelector labels.Selector) cache.Cache {
+	c, err := cache.New(p.GetRESTConfig(), cache.Options{
+		Scheme:          p.scheme,
+		Mapper:          p.GetClient().RESTMapper(),
+		DefaultSelector: cache.ObjectSelector{Label: labelSelector},
+		Namespace:       namespace,
```

Comment:
Refactored to use this GetCache function.

A question @sbueringer / @vincepri : would it be better to have a single Cache for the `clusterProxy`, and always return a new informer from this single cache?

This would require filtering for the labels and namespace inside `AddFunc` / `UpdateFun`.

Reply:
I think it would be better to have a single cache. And yup I think you're right this requires filterin gin AddFunc / UpdateFunc.

The nice thing should be that there is only one underlying informer (Get should just return the existing one)

Reply:
Sorry just now looking at the code.

I was thinking more about creating a cache singleton without labelselector / namespace which is returned every time.

Then consumers can call GetCache() and then GetInformer() then AddEventHandler (or the one with sync period)

Otherwise we get one cache and I guess a separate informer per deployment

Reply:
Re-reading. Ah that was probably what your question was aiming for, so I guess my first answer was okay :)

Reply:
Jep, already refactored for that :-)

Reply:
Ah, also adding the informer as singleton 👍

Reply:
I think that's how cache just works internally